esphome/components/esp32_touch/esp32_touch.cpp

```diff
@@ -142,6 +169,24 @@ void ESP32TouchComponent::on_shutdown() {
 
 ESP32TouchBinarySensor::ESP32TouchBinarySensor(const std::string &name, touch_pad_t touch_pad, uint16_t threshold)
     : BinarySensor(name), touch_pad_(touch_pad), threshold_(threshold) {}
+ESP32TouchBinarySensor::ESP32TouchBinarySensor(const std::string &name, touch_pad_t touch_pad, uint16_t offset,
+                                               uint16_t interval, uint16_t sample_size)
+    : BinarySensor(name),
+      touch_pad_(touch_pad),
+      offset_(offset),
+      interval_(interval * 1000),
```

Comment:
```suggestion
      interval_(interval),
```

There's no reason we need to constrain this value to whole seconds. Then also change the validator to `cv.positive_time_period_milliseconds`

Reply:
True, can do